core/src/main/clojure/xtdb/log.clj

```diff
@@ -226,7 +223,7 @@
         (.writeObject query-writer sql)
 
         (when-let [arg-rows (.argRows op)]
```

Comment:
`not-empty` call here around `argRows` would alleviate the `apply max` concern above